osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/application/tasks/ScheduledTaskExecutorService.java

```diff
@@ -41,11 +45,47 @@ public class ScheduledTaskExecutorService {
   @Autowired private ScheduledTaskExecutorJobConfig scheduledTaskExecutorJobConfig;
 
   public void processScheduledTasks() {
+    this.processStrandedScheduledTasks();
     this.processScheduledTasks(ScheduledTaskStatusType.NEW);
     this.processScheduledTasks(ScheduledTaskStatusType.RETRY);
   }
 
+  private void processStrandedScheduledTasks() {
+    final List<ScheduledTask> scheduledTasks =
+        this.getScheduledTasks(ScheduledTaskStatusType.PENDING);
+    final List<ScheduledTask> strandedScheduledTasks =
+        scheduledTasks.stream()
+            .filter(
+                st ->
+                    st.getModificationTimeInstant()
+                        .isBefore(
+                            Instant.now()
+                                .minus(
+                                    this.scheduledTaskExecutorJobConfig
+                                        .scheduledTaskPendingDurationMaxSeconds(),
+                                    ChronoUnit.SECONDS)))
```

Comment:
I think extracting this to a method would make it more readable.

Reply:
seperated into 4 statement with explanatory variable names.